src/App.js

```diff
@@ -1,25 +1,163 @@
+
+    this.setState({startDate : firstDay});
+    this.setState({endDate : lastDay});
+  }
+
+  setStateDateFromYear = (year) => {
+    let firstDay = new Date(year, 0, 1);
+    let lastDay = new Date(year, 11, 31);
+    this.setState({year : year});
+    this.setState({month1 : 0});
+    this.setState({month2 : 11});
+    this.setState({startDate : firstDay});
+    this.setState({endDate : lastDay});
+    this.setState({selectAll : false});
+  }
+
+  setStateDateFromQuarter = (month1, month2) => {
+    let firstDay = new Date(this.state.year, month1, 1);
+    let lastDay = new Date(this.state.year, month2 + 1, 0);
+
+    /*if(month1<=this.state.month1 || month2===this.state.month2){
+      this.setState({month1 : month1});
+      this.setState({month2 : month2});
+    }else{
+      this.setState({month2 : month2});
+    }*/
+
+    this.setState({month1 : month1});
+    this.setState({month2 : month2});
+
+    this.setState({startDate : firstDay});
+    this.setState({endDate : lastDay});
+    this.setState({selectAll : false});
+  }
+
+  setStateDateFromMonth = (month) => {
+    let firstDay = new Date(this.state.year, month, 1);
+    let lastDay = new Date(this.state.year, month + 1, 0);
+    if(month<=this.state.month1 || month===this.state.month2){
+      this.setState({month1 : month});
+      this.setState({month2 : month});
+      this.setState({startDate : firstDay});
+    }else{
+      this.setState({month2 : month});
+    }
+    this.setState({endDate : lastDay});
+    this.setState({selectAll : false});
+  }
+
+  selectedMode=(mode)=>{
+    switch(mode){
+      case "Yearly":
+        return <CalendarYearly 
+        year={this.state.year} 
+        month1={this.state.month1}
+        month2={this.state.month2}
+        yearLowest={this.state.yearLowerLimit}
+        yearAmmount={this.state.yearAmmountLimit}
+        selectAll={this.state.selectAll}
+        setDates={this.setStateDateFromYear}
+        />;
+      case "Quarterly":
+        return <CalendarQuarterly 
+        selectAll={this.state.selectAll}  
+        year={this.state.year} 
+        month1={this.state.month1}
+        month2={this.state.month2}
+        setYear={this.setStateYear}
+        setDates={this.setStateDateFromQuarter}
+        />;
+      case "Monthly": 
+        return <CalendarMonthly 
+        selectAll={this.state.selectAll}  
+        year={this.state.year}
+        month1={this.state.month1}
+        month2={this.state.month2}
+        setYear={this.setStateYear}
+        setDates={this.setStateDateFromMonth}
+        />;
+      }
+    }
+
+  render(){
+    //console.log(this.state.eventList[0]); 
+
+    return (
+      <div className={styles.App}>
+        
+        <DateRangePicker 
+        setPeriod={this.setStatePeriod}
+        selectAll={this.state.selectAll}  
+        setSelectAll={this.setStateSelectAll}
+        selectAllenabled={this.state.period!="Yearly"}
+        dateStart={this.state.startDate}
+        dateEnd={this.state.endDate}
```

Comment:
@edwartexe be aware of spaces and code readability
```suggestion
          setPeriod={this.setStatePeriod}
          selectAll={this.state.selectAll}  
          setSelectAll={this.setStateSelectAll}
          selectAllenabled={this.state.period != 'Yearly'}
          dateStart={this.state.startDate}
          dateEnd={this.state.endDate}
```